### PR TITLE
Fix incorrect indentation on Conditional-expr formatting when written over mutiple lines

### LINKED ISSUE #35163: Conditional-expr formatting does not give the correct indentations when written over mutiple lines
**Description:**
$subject.

When the conditional expression is lengthy, it is very common to write it in the following forms. But the formatter does not provide correct indentations in these scenarios.

```
    // Scenario 1
    TypeDescriptor variableName = expression1 ?
            expresssion2 : expression3;

    // Scenario 2
    TypeDescriptor variableName = expression1 ? expresssion2 :
            expression3;

    // Scenario 3
    TypeDescriptor variableName = expression1 ?
            expresssion2 :
            expression3;
```

Cosider the following sample.
```bal
public function main() {
// Scenario 1
TypeDescriptor variableName = expression1 ?
expresssion2 : expression3;

// Scenario 2
TypeDescriptor variableName = expression1 ? expresssion2 :
expression3;

// Scenario 3
TypeDescriptor variableName = expression1 ?
expresssion2 :
expression3;
}
```

It gets formatted as below.
```bal
public function main() {
    // Scenario 1
    TypeDescriptor variableName = expression1 ?
expresssion2 : expression3;

    // Scenario 2
    TypeDescriptor variableName = expression1 ? expresssion2 :
expression3;

    // Scenario 3
    TypeDescriptor variableName = expression1 ?
expresssion2 :
expression3;
}
```
**Affected Versions:**
2201.0.0

### PR DESCRIPTION
**Pull request by AzeemMuzammil**
Fixes #35163

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [x] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/formatter/modules/formatter-core/src/test/resources/expressions/conditional/assert/conditional_expression_1.bal [hunk 1]
```diff
@@ -5,4 +5,14 @@ function getValue(int v) returns int|() {
...
+
+    string var1 = true ?
+        "hello" : "hi";
+
+    string var2 = true ? "hello" :
+        "hi";
+
+    string var3 = true ?
+        "hello" :
+        "hi";
```
Comment (NipunaRanasinghe) [inline misc/formatter/modules/formatter-core/src/test/resources/expressions/conditional/assert/conditional_expression_1.bal:17]: Shall we add a test case for the scenario where only the end expression goes to a new line?

e.g.

```
string var3 = true ? "hello" :
        "hi";
```

and 
```
string var3 = true ? "hello" 
        : "hi";
```

Comment (AzeemMuzammil) [reply misc/formatter/modules/formatter-core/src/test/resources/expressions/conditional/assert/conditional_expression_1.bal:17]: +1

**Review by NipunaRanasinghe — DISMISSED** [review]
LGTM

**Comment by AzeemMuzammil** [comment]
Closing this PR with #37788

**Pull request closed without merging**